apps/designer/app/designer/shared/assets/update-state-assets.ts

```diff
@@ -0,0 +1,56 @@
+import deepEqual from "fast-deep-equal";
+import type { Asset } from "@webstudio-is/asset-uploader";
+import { DeletingAsset, PreviewAsset } from "./types";
+
+type StateAssets = Asset | PreviewAsset | DeletingAsset;
```

Comment:
Strange name `StateAssets`. What does prefix `State` means? And why plural Asset**s**?

Names that would make more sense to me: `AssetWithState`, `LocalAsset`, `ClientSideAsset`, `ClientAsset`